### PR TITLE
Bump `nginx-ingress-controller-seed` to `v1.3.0` for `1.22+` seeds

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
Bump `nginx-ingress-controller-seed` to `v1.3.0` for `1.22+` seeds.

**Special notes for your reviewer**:
/cc @shafeeqes 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/component/nginxingress/nginxingress.go [hunk 1]
```diff
@@ -308,6 +308,11 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 					Resources: []string{"ingressclasses"},
 					Verbs:     []string{"get", "list", "watch"},
 				},
+				{
+					APIGroups: []string{"coordination.k8s.io"},
+					Resources: []string{"leases"},
+					Verbs:     []string{"list", "watch"},
+				},
```
Comment (acumino) [inline pkg/operation/botanist/component/nginxingress/nginxingress.go:326]: This is being added for all Kubernetes versions whereas it is required only for Kubernetes version >=1.22.

Comment (rfranzke) [reply pkg/operation/botanist/component/nginxingress/nginxingress.go:326]: True, but I don't consider this an issue (we already do it similarly for other resources, e.g. https://github.com/gardener/gardener/blob/b93eeced331603a6527d71555fa5d05093852b42/pkg/operation/botanist/component/nginxingress/nginxingress.go#L306-L310)

**Review by shafeeqes — COMMENTED** [review]
Thanks for the PR!
On testing, with this change alone, the lease is not getting created.
We might need to adapt https://github.com/gardener/gardener/blob/59f35eab004b3128d8a6f21b11decd5872b57d43/pkg/operation/botanist/component/nginxingress/nginxingress.go#L221 with `create` permissions for `lease`s.
Ref: https://github.com/kubernetes/ingress-nginx/pull/8733/commits/62a876a80875d53756b0831f03d518295f76f9f8

**Comment by shafeeqes** [comment]
Lease is getting created and both `configmap` and `lease` contain the same holder identity.
/lgtm

**Review by acumino — APPROVED** [review]
/lgtm
/approve

**Pull request merged**